### ISSUE TITLE
java.lang.string cannot be cast to com.facebook.react.uimanager.Accessibility DelegateUtil$AccessibilityRole

### ISSUE THREAD

**Issue by MickeyMouse623** [facebook/react-native#21881]
## Environment
Run `react-native info` in your terminal and paste its contents here.
![image](https://user-images.githubusercontent.com/18524460/47268567-3f0d4100-d585-11e8-8d2c-813528ca7527.png)

```javascript
"dependencies": {
    "antd-mobile-rn": "^2.2.1",
    "axios": "^0.18.0",
    "react": "16.6.0-alpha.8af6728",
    "react-native": "0.57.3",
    "react-native-swiper": "^1.5.13",
    "react-native-vector-icons": "^6.0.2",
    "react-navigation": "^2.17.0",
    "react-redux": "^5.0.7",
    "redux": "^4.0.1"
  },
  "devDependencies": {
    "babel-jest": "23.6.0",
    "babel-plugin-import": "^1.10.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.48.1",
    "react-test-renderer": "16.6.0-alpha.8af6728"
  },
```
## Description
事故过程描述：
1、首先我在 `真机` 上删除的 `APP`
2、随后执行 `react-native run-android`，这时无法将 `App` 推送到 `真机设备` 上安装，出现底部的错误提示
```javascript
    D:\work\xlz\yaruiApp>react-native run-android
Starting JS server...
Building and installing the app on the device (cd android && gradlew.bat installDebug)...

> Configure project :app
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> Task :app:installDebug
Installing APK 'app-debug.apk' on 'MI 5 - 8.0.0' for app:debug
Installed on 1 device.


BUILD SUCCESSFUL in 18s
46 actionable tasks: 9 executed, 37 up-to-date
Running C:\Users\admin\AppData\Local\Android\Sdk/platform-tools/adb -s aba9a91b reverse tcp:8081 tcp:8081
Starting the app on aba9a91b (C:\Users\admin\AppData\Local\Android\Sdk/platform-tools/adb -s aba9a91b shell am start -n com.yaruiapp/com.yaruiapp.MainActivity)...
Starting: Intent { cmp=com.yaruiapp/.MainActivity }
Error type 3
Error: Activity class {com.yaruiapp/com.yaruiapp.MainActivity} does not exist.
```
同一套代码，在 `"react": "^16.5.2"` 和 ` "react-native": "0.57.2"` 版本上并没有此报错。

3、查阅度娘后，修改了 `/android/app/build.gradle` 中的 `applicationId` 的值 `com.yaruiapp`，修改为 `com.xlz.yaruiapp`

4、最后可以将 `App` 安装至 `真机设备`，但出现了 `title` 中描述的问题

总结：两个问题
问题一：
``` javascript
Error type 3
Error: Activity class {com.yaruiapp/com.yaruiapp.MainActivity} does not exist.
```

问题二：
```javascript
java.lang.string cannot be cast to com.facebook.react.uimanager.Accessibility DelegateUtil$AccessibilityRole
```
